monitoring/testonly/delta.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testonly
+
+import (
+	"github.com/google/trillian/monitoring"
+)
+
+// CounterSnapshot records the values of a counter at a specific point in time.
+type CounterSnapshot struct {
+	c      monitoring.Counter
+	values map[string]float64
+}
+
+// SnapshotCounter records the current values of a counter.
+// It will only record values associated with the given labels.
+func SnapshotCounter(c monitoring.Counter, labels ...string) CounterSnapshot {
```

Comment:
It looks a bit confusing when I compare `SnapshotCounter` to a regular `Counter`. Both accept a list of string `labels`, but `Counter` treats them as labels of a single multi-label metric, while `SnapshotCounter` treats them as labels of multiple single-label metrics.

To make it more general and avoid this confusion might be worth changing the signature to:
```
func NewCounterSnapshot(c monitoring.Counter, labels ...[]string) CounterSnapshot
```
... and documenting what `labels` actually encodes.

Note that I also suggest renaming the function.

Reply:
I don't feel strongly about renaming though, `SnapshotCounter` seems like a good verb.

Reply:
Thanks for the explanation, I misunderstood how those labels were used in the monitoring package. I've fixed this PR to use them as intended.